ckanext/unaids/plugin.py

```diff
@@ -208,8 +207,9 @@ def after_update(self, context, pkg_dict):
                     dataset_id=pkg_dict["id"],
                     recipient_org_id=org_to_allow_transfer_to[0],
                 )
-        if pkg_dict.get("validate_package") and not context.get("_dont_validate"):
-            logging.warning("VALIDATING ENTIRE PACKAGE")
+
+        if pkg_dict['id'] in self.resources_to_validate_package and not context.get("_dont_validate"):
```

Comment:
If user passes `context["_dont_validate"]` we skip the whole dataset validation but shouldn't you clear `self.resources_to_validate_package[pkg_dict['id']]` so that the validation is skipped permanently (and not repeated next time dataset is updated)?